File: packages/ls-react-native-crypto/android/src/main/java/com/lightspark/sdk/rn/crypto/ExportedKeys.kt
package com.lightspark.sdk.rn.crypto

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record
import expo.modules.kotlin.types.Enumerable

class ExportedKey(
    @Field
    val keyBytes: String = "",
    @Field
    val format: KeyFormat = KeyFormat.PKCS8,
) : Record {
    enum class KeyFormat(val value: String) : Enumerable {
        PKCS8("pkcs8"),
        SPKI("spki"),
        PKS1("pks1"),
        ;
    }
}

class ExportedKeys(
    @Field
    val publicKey: ExportedKey = ExportedKey(),
    @Field
    val privateKey: ExportedKey = ExportedKey(),
    @Field
    val alias: String = "",
) : Record

File: packages/ls-react-native-crypto/ios/ExportedKeys.swift
import ExpoModulesCore

internal struct ExportedKey: Record {
  @Field
  var format: Format = .pkcs8

  @Field
  var keyBytes: String

  internal enum Format: String {
    case pkcs8 = "pkcs8"
    case spki = "spki"
    case pkcs1 = "pkcs1"
  }
}

internal struct ExportedKeys: Record {
  @Field
  var privateKey: ExportedKey

  @Field
  var publicKey: ExportedKey

  @Field
  var alias: String
}
